src/include/miopen/tensor/problem_description.hpp

```diff
@@ -0,0 +1,152 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <miopen/problem_description_base.hpp>
+#include <miopen/tensor.hpp>
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace tensor {
+
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription(const miopenTensorOp_t tensorOp_,
+                       const void* alpha0_,
+                       const void* alpha1_,
+                       const void* beta_,
+                       const TensorDescriptor& aTensorDesc_,
+                       const TensorDescriptor& bTensorDesc_,
+                       const TensorDescriptor& cTensorDesc_,
+                       const bool nonStandardSquash_)
+        : tensorOp(tensorOp_),
+          aTensorDesc(aTensorDesc_),
+          bTensorDesc(bTensorDesc_),
+          cTensorDesc(cTensorDesc_),
+          nonStandardSquash(nonStandardSquash_)
+    {
+        CheckAndAssignAlphaBeta(alpha0_, alpha1_, beta_);
+
+        if(aTensorDesc.GetElementSize() != cTensorDesc.GetElementSize())
+        {
+            MIOPEN_THROW("A and C Tensors do not match");
+        }
+
+        if(bTensorDesc.GetType() != cTensorDesc.GetType())
+        {
+            MIOPEN_THROW("Datatypes for B and C tensors do not match !");
+        }
+
+        auto blens = bTensorDesc.GetLengths();
+        auto clens = cTensorDesc.GetLengths();
+        if(clens.size() > 5)
+        {
+            MIOPEN_THROW("Tensor dimension larger than 5: " + std::to_string(clens.size()));
+        }
+
+        if(blens.size() != clens.size())
+        {
+            MIOPEN_THROW("Number of dims in B and C Tensors do not match: " +
+                         std::to_string(blens.size()) + ", " + std::to_string(clens.size()));
+        }
+
+        if(!nonStandardSquash)
+        {
+            for(std::size_t i = 0; i < clens.size(); i++)
```

Comment:
https://en.cppreference.com/w/cpp/algorithm/all_any_none_of
this would make it a one-liner with clear intent

Reply:
what is your idea on how to use any of those functions for this? I think that it is hard to use those with two iterators etc. so maybe this old implementation is better